scalafix-cli/src/main/scala/scalafix/internal/interfaces/ScalafixArgumentsImpl.scala

```diff
@@ -101,12 +101,14 @@ final case class ScalafixArgumentsImpl(args: Args = Args.default)
           dependency.getVersion,
           Versions.nightly
         ) match {
-          case Incompatible =>
-            throw new ScalafixException(
-              s"Scalafix version ${Versions.nightly} cannot load the registered external rules, " +
-                s"please upgrade to ${dependency.getVersion} or later"
+          case TemptativeUp(compatibleRunWith) =>
+            args.out.println(
+              s"""Loading external rule(s) built against a recent version of Scalafix (${dependency.getVersion}).
+                |This might not be a problem, but in case you run into unexpected behavior, you
+                |should upgrade Scalafix to ${compatibleRunWith}.
+              """.stripMargin
```

Comment:
The warn messages are super clear! 
We can definitely turn that to an exception if many users complains that with scalafix 0.10.x, rules in 0.10.x x>0, it breaks. It would mean we don't pay too much attention to warn messages and in this case an exception is better.

Reply:
> We can definitely turn that to an exception if many users complains that with scalafix 0.10.x, rules in 0.10.x x>0, it breaks

What do you mean? 0.10.x runtimes will already be in the wild. we can't retroactively turn the warning into an exception.

Reply:
What I mean, we can still turn it into an exception in 0.10.5 for example, if we think a warning is not enough. I don't know if it's ok to that. 


This message is only displayed when we have a tentativeUp starting `0.10.0`. 
The only case will be: A rule is published with `0.10.1` and the user has upgraded the rule, but not the scalafix version which is still `0.10.0`.
Maybe throwing an exception is ok in that case.

Reply:
> What I mean, we can still turn it into an exception in 0.10.5 for example, if we think a warning is not enough. I don't know if it's ok to that.

We can, but we would want to be more verbose in case 0.10.5 (or 0.11.0) brings silent changes important enough to prevent users to use an earlier version to run rules built/tested against it. But it would be too late as 0.10.4 and earlier are out!

Reply:
As discussed, let's for go for warn now. We can revisit for 1.x, maybe with an exception if build is newer by major or minor only, keeping a warn for patch diff.